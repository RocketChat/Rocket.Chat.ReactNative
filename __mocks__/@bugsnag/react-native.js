export default {
	start: () => '',
	leaveBreadcrumb: () => '',
	notify: () => ''
};
