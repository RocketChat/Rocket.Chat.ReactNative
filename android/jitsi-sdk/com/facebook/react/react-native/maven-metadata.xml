<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>0.59.10</release>
    <versions>
      <version>0.59.10</version>
    </versions>
    <lastUpdated>20190910182229</lastUpdated>
  </versioning>
</metadata>
