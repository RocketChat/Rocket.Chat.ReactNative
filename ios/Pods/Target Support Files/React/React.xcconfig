CLANG_CXX_LANGUAGE_STANDARD = c++14
CONFIGURATION_BUILD_DIR = ${PODS_CONFIGURATION_BUILD_DIR}/React
GCC_PREPROCESSOR_DEFINITIONS = $(inherited) COCOAPODS=1
HEADER_SEARCH_PATHS = $(inherited) "${PODS_ROOT}/Headers/Private" "${PODS_ROOT}/Headers/Private/React" "${PODS_ROOT}/Headers/Public" "${PODS_ROOT}/Headers/Public/React" "${PODS_ROOT}/Headers/Public/yoga" "$(PODS_TARGET_SRCROOT)/ReactCommon"
PODS_BUILD_DIR = ${BUILD_DIR}
PODS_CONFIGURATION_BUILD_DIR = ${PODS_BUILD_DIR}/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)
PODS_ROOT = ${SRCROOT}
PODS_TARGET_SRCROOT = ${PODS_ROOT}/../../node_modules/react-native
PRODUCT_BUNDLE_IDENTIFIER = org.cocoapods.${PRODUCT_NAME:rfc1034identifier}
SKIP_INSTALL = YES
