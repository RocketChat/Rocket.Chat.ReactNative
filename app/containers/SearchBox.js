import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';
import Touchable from 'react-native-platform-touchable';

import TextInput from '../presentation/TextInput';
import I18n from '../i18n';
import { CustomIcon } from '../lib/Icons';
import sharedStyles from '../views/Styles';
import { withTheme } from '../theme';
import { themes } from '../constants/colors';
import { isIOS } from '../utils/deviceInfo';

const styles = StyleSheet.create({
	container: {
		flexDirection: 'row',
		alignItems: 'center',
		flex: 1
	},
	searchBox: {
		alignItems: 'center',
		borderRadius: 10,
		flexDirection: 'row',
		fontSize: 17,
		height: 36,
		margin: 16,
		marginVertical: 10,
		paddingHorizontal: 10,
		flex: 1
	},
	input: {
		flex: 1,
		fontSize: 17,
		marginLeft: 8,
		paddingTop: 0,
		paddingBottom: 0,
		...sharedStyles.textRegular
	},
	cancel: {
		marginRight: 15
	},
	cancelText: {
		...sharedStyles.textRegular,
		fontSize: 17
	}
});

const CancelButton = (onCancelPress, theme) => (
	<Touchable onPress={onCancelPress} style={styles.cancel}>
		<Text style={[styles.cancelText, { color: themes[theme].tintColor }]}>{I18n.t('Cancel')}</Text>
	</Touchable>
);

const SearchBox = ({
	onChangeText, onSubmitEditing, testID, hasCancel, onCancelPress, inputRef, theme, ...props
}) => (
	<View
		style={[
			styles.container,
			{ backgroundColor: isIOS ? themes[theme].headerBackground : themes[theme].headerSecondaryBackground }
		]}
	>
		<View style={[styles.searchBox, { backgroundColor: themes[theme].searchboxBackground }]}>
			<CustomIcon name='magnifier' size={14} color={themes[theme].auxiliaryText} />
			<TextInput
				ref={inputRef}
				autoCapitalize='none'
				autoCorrect={false}
				blurOnSubmit
				clearButtonMode='while-editing'
				placeholder={I18n.t('Search')}
				returnKeyType='search'
				style={styles.input}
				testID={testID}
				underlineColorAndroid='transparent'
				onChangeText={onChangeText}
				onSubmitEditing={onSubmitEditing}
				theme={theme}
				{...props}
			/>
		</View>
		{ hasCancel ? CancelButton(onCancelPress, theme) : null }
	</View>
);

SearchBox.propTypes = {
	onChangeText: PropTypes.func.isRequired,
	onSubmitEditing: PropTypes.func,
	hasCancel: PropTypes.bool,
	onCancelPress: PropTypes.func,
	theme: PropTypes.string,
	inputRef: PropTypes.func,
	testID: PropTypes.string
};

export default withTheme(SearchBox);
