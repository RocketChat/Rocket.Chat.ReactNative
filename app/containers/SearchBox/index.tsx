import React, { useCallback, useState } from 'react';
import { StyleSheet, TextInputProps, View } from 'react-native';

import { useTheme } from '../../theme';
import I18n from '../../i18n';
import { FormTextInput } from '../TextInput';

const styles = StyleSheet.create({
	inputContainer: {
		margin: 16,
		marginBottom: 16
	}
});

const SearchBox = ({ onChangeText, onSubmitEditing, testID }: TextInputProps): JSX.Element => {
	const [text, setText] = useState('');

	const { colors } = useTheme();

	const internalOnChangeText = useCallback(value => {
		setText(value);
		onChangeText?.(value);
	}, []);

	return (
		<View testID='searchbox' style={{ backgroundColor: colors.backgroundColor }}>
			<FormTextInput
				autoCapitalize='none'
				autoCorrect={false}
				blurOnSubmit
				placeholder={I18n.t('Search')}
				returnKeyType='search'
				underlineColorAndroid='transparent'
				containerStyle={styles.inputContainer}
				onChangeText={internalOnChangeText}
				onSubmitEditing={onSubmitEditing}
				value={text}
				testID={testID}
				onClearInput={() => internalOnChangeText('')}
				iconRight={'search'}
			/>
		</View>
	);
};

export default SearchBox;
