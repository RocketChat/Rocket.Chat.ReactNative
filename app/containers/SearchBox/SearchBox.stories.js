import { storiesOf } from '@storybook/react-native';
import React from 'react';
import SearchBox from './index';

const stories = storiesOf('SearchBox', module);

stories.add('Item', () => <SearchBox />);
