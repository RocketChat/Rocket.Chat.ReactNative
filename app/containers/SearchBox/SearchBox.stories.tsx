import React from 'react';

import SearchBox from './index';

export default {
	title: 'SearchBox'
};

export const Basic = () => <SearchBox />;
