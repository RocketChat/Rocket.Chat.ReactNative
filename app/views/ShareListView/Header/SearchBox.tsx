import React from 'react';
import { StyleSheet, Text, TextInput as RNTextInput, TextInputProps, View } from 'react-native';
import Touchable from 'react-native-platform-touchable';

import { themes } from '../../../lib/constants';
import I18n from '../../../i18n';
import { CustomIcon } from '../../../containers/CustomIcon';
import { TextInput } from '../../../containers/TextInput';
import { useTheme } from '../../../theme';
import { isIOS } from '../../../lib/methods/helpers';
import sharedStyles from '../../Styles';

const styles = StyleSheet.create({
	container: {
		flexDirection: 'row',
		alignItems: 'center',
		flex: 1
	},
	searchBox: {
		alignItems: 'center',
		borderRadius: 10,
		flexDirection: 'row',
		fontSize: 17,
		height: 36,
		margin: 16,
		marginVertical: 10,
		paddingHorizontal: 10,
		flex: 1
	},
	input: {
		flex: 1,
		fontSize: 17,
		marginLeft: 8,
		paddingTop: 0,
		paddingBottom: 0,
		...sharedStyles.textRegular
	},
	cancel: {
		marginRight: 15
	},
	cancelText: {
		...sharedStyles.textRegular,
		fontSize: 17
	}
});

interface ISearchBox extends TextInputProps {
	value?: string;
	hasCancel?: boolean;
	onCancelPress?: () => void;
	inputRef?: React.Ref<RNTextInput>;
}

const CancelButton = ({ onCancelPress }: { onCancelPress?: () => void }) => {
	const { theme } = useTheme();
	return (
		<Touchable onPress={onCancelPress} style={styles.cancel}>
			<Text style={[styles.cancelText, { color: themes[theme].headerTintColor }]}>{I18n.t('Cancel')}</Text>
		</Touchable>
	);
};

const SearchBox = ({ hasCancel, onCancelPress, inputRef, ...props }: ISearchBox): React.ReactElement => {
	const { theme } = useTheme();
	return (
		<View
			style={[
				styles.container,
				{ backgroundColor: isIOS ? themes[theme].headerBackground : themes[theme].headerSecondaryBackground }
			]}
		>
			<View style={[styles.searchBox, { backgroundColor: themes[theme].searchboxBackground }]}>
				<CustomIcon name='search' size={14} color={themes[theme].auxiliaryText} />
				<TextInput
					ref={inputRef}
					autoCapitalize='none'
					autoCorrect={false}
					blurOnSubmit
					clearButtonMode='while-editing'
					placeholder={I18n.t('Search')}
					returnKeyType='search'
					style={styles.input}
					underlineColorAndroid='transparent'
					{...props}
				/>
			</View>
			{hasCancel && onCancelPress ? <CancelButton onCancelPress={onCancelPress} /> : null}
		</View>
	);
};

export default SearchBox;
