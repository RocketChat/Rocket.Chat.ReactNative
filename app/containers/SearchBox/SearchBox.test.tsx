import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import SearchBox from '.';

const onChangeTextMock = jest.fn();

const testSearchInputs = {
	onChangeText: onChangeTextMock,
	testID: 'search-box-text-input'
};

describe('SearchBox', () => {
	it('should render the searchbox component', () => {
		const { findByTestId } = render(<SearchBox onChangeText={testSearchInputs.onChangeText} testID={testSearchInputs.testID} />);

		expect(findByTestId('searchbox')).toBeTruthy();
	});

	it('should not render clear-input icon', async () => {
		const { queryByTestId } = render(<SearchBox onChangeText={testSearchInputs.onChangeText} testID={testSearchInputs.testID} />);
		const clearInput = await queryByTestId('clear-text-input');
		expect(clearInput).toBeNull();
	});

	it('should input new value with onChangeText function', async () => {
		const { findByTestId } = render(<SearchBox onChangeText={onChangeTextMock} testID={testSearchInputs.testID} />);

		const component = await findByTestId(testSearchInputs.testID);
		fireEvent.changeText(component, 'new-input-value');
		expect(onChangeTextMock).toHaveBeenCalledWith('new-input-value');
	});

	it('should clear input when clear icon is pressed', async () => {
		const { findByTestId } = render(<SearchBox onChangeText={onChangeTextMock} testID={testSearchInputs.testID} />);

		const component = await findByTestId(testSearchInputs.testID);
		fireEvent.changeText(component, 'new-input-value');

		const clearTextInput = await findByTestId('clear-text-input');
		fireEvent.press(clearTextInput);
		expect(onChangeTextMock).toHaveBeenCalledWith('');
	});
});
