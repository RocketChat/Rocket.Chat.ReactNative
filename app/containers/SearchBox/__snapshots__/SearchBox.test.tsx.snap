// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Basic should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
  testID="searchbox"
>
  <View>
    <View>
      <View
        accessibilityLabel=""
        accessible={true}
        style={
          [
            {
              "gap": 4,
              "marginBottom": 10,
            },
            {
              "margin": 16,
              "marginBottom": 16,
            },
          ]
        }
      >
        <View
          accessible={true}
          style={
            {
              "justifyContent": "center",
              "position": "relative",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            blurOnSubmit={true}
            keyboardAppearance="light"
            onChangeText={[Function]}
            placeholder="Search"
            placeholderTextColor="#9EA2A8"
            returnKeyType="search"
            style={
              [
                {
                  "color": "#1F2329",
                },
                [
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "fontFamily": "Inter",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "paddingHorizontal": 16,
                    "paddingVertical": 14,
                    "textAlign": "left",
                  },
                  undefined,
                  {
                    "paddingRight": 45,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#9EA2A8",
                    "color": "#1F2329",
                  },
                  {},
                  undefined,
                ],
                {
                  "textAlign": "auto",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            accessible={false}
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#2F343D",
                  "fontSize": NaN,
                },
                [
                  {
                    "lineHeight": NaN,
                  },
                  [
                    {
                      "position": "absolute",
                    },
                    {
                      "right": 12,
                    },
                  ],
                ],
                {
                  "fontFamily": "custom",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î¦¯
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
